gcsfs/core.py

```diff
@@ -296,13 +329,47 @@ def _call(self, method, path, *args, **kwargs):
         meth = getattr(requests, method)
         if args:
             path = path.format(*[quote_plus(p) for p in args])
-        r = meth(self.base + path, headers=self.header, params=kwargs,
-                 json=json)
+        for retry in range(RETRY_COUNT):
+            try:
+                time.sleep(2**retry-1)
+                r = meth(self.base + path, headers=self.header, params=kwargs,
+                         json=json)
+                validate_response(r, path)
+                break
+            except RuntimeError as e:
+                warnings.warn("_call caught RuntimeError: " + e.__repr__())
+                with open('/tmp/error', 'wb') as f:
+                    pickle.dump(e, f)
+                print('error saved to /tmp/error')
+                if retry + 1 == RETRY_COUNT:
+                    warnings.warn('GCS call failure, out of retries: ' + str(e))
+                    raise e
+                html_error = _asHtmlError(e)
```

Comment:
It seems like a lot of code to get the HTTP code. Can you use `r.status_code`, which is always an integer?